src/GitHub.TeamFoundation.14/Services/VSGitServices.cs

```diff
@@ -81,25 +83,64 @@ public string GetLocalClonePathFromGitProvider()
             bool recurseSubmodules,
             object progress = null)
         {
-#if TEAMEXPLORER14
-            var gitExt = serviceProvider.GetService<IGitRepositoriesExt>();
-            gitExt.Clone(cloneUrl, clonePath, recurseSubmodules ? CloneOptions.RecurseSubmodule : CloneOptions.None);
+            var teamExplorer = serviceProvider.TryGetService<ITeamExplorer>();
+            Assumes.Present(teamExplorer);
 
-            // The operation will have completed when CanClone goes false and then true again.
-            await gitExt.WhenAnyValue(x => x.CanClone).Where(x => !x).Take(1);
-            await gitExt.WhenAnyValue(x => x.CanClone).Where(x => x).Take(1);
+#if TEAMEXPLORER14
```

Comment:
So this is the new bit right? So instead of waiting for the response from TE, we're starting the clone and immediately navigating to the TE Home page and then the user will see the TE change as a result of the clone completing but TE is handling that. And we can identify that change (instead of actually waiting for an update from TE)

Reply:
We can infer that the clone operation has completed when the progress bar is hidden and automatically change context to the newly cloned repository. If we didn't change repository context at this point, the user would know the clone had completed, but they'd be left to find and open a solution from the newly cloned folder (not a nice experience).